website/docs/cdktf/cli-reference/commands.mdx

```diff
@@ -676,3 +676,52 @@ Add a provider from a private registry to the project.
 ```bash
 $ cdktf provider add registry.example.com/acme/my-provider
 ```
+
+## provider add
+
+This command facilitates upgrading Terraform providers in a CDKTF project. If a [pre-built provider](/cdktf/concepts/providers#install-pre-built-providers) for the requested provider is installed we will update it to the newest version matching the version constraint and your `cdktf` version. Otherwise, the provider will be added to the `cdktf.json` config and `cdktf get` will be automatically invoked to generate [local provider bindings](/cdktf/concepts/providers#add-provider-to-cdktf-json).
+
+```sh
+$ cdktf provider add --help
+```
+
+**Help Output**
+
+```
+cdktf provider upgrade <provider...>
+
+Upgrade one or more Terraform providers in your project.
+
+Positionals:
+  provider  Name of the provider to upgrade. Can include a version constraint (e.g. aws@~>4.0).
+                                                                     [array] [required] [default: []]
+
+Options:
+      --version                   Show version number                                       [boolean]
+      --disable-plugin-cache-env  Dont set TF_PLUGIN_CACHE_DIR automatically. This is useful when the
+                                  plugin cache is configured differently. Supported using the env
+                                  CDKTF_DISABLE_PLUGIN_CACHE_ENV.          [boolean] [default: false]
+      --log-level                 Which log level should be written. Only supported via setting the
+                                  env CDKTF_LOG_LEVEL                                        [string]
+  -h, --help                      Show help                                                 [boolean]
+```
+
+**Examples**
+
+Upgrade the aws provider to the newest version available for your `cdktf` version.
+
+```bash
+$ cdktf provider add aws
+```
+
+Upgrade the aws provider to the newest version available for your `cdktf` version and at version `4` of the provider version.
```

Comment:
What happens if your cdktf version doesn't support the constraint you added - like what if in this case your cdktf version doesn't support aws version 4?

Reply:
CDKTF support all provider versions. If there is no pre-built provider we will generate bindings locally through cdktf get